deblur/workflow.py

```diff
@@ -382,40 +346,133 @@ def write_biom_table(table, biom_fp):
     biom_fp: string
         filepath to output BIOM table
     """
+    logger = logging.getLogger(__name__)
+    logger.debug('write_biom_table to file %s' % biom_fp)
     with biom_open(biom_fp, 'w') as f:
         if HAVE_H5PY:
+            logger.debug('saving with h5py')
             table.to_hdf5(h5grp=f, generated_by="deblur")
+            logger.debug('wrote to hdf5 file %s' % biom_fp)
         else:
+            logger.debug('no h5py. saving to json')
             table.to_json(direct_io=f, generated_by="deblur")
+            logger.debug('wrote to json file %s' % biom_fp)
+
 
+def get_files_for_table(input_dir,
+                        file_end='.fasta.trim.derep.no_artifacts.msa.deblur.no_chimeras'):
+    """Get a list of files to add to the output table
 
-def merge_otu_tables(output_fp, all_tables):
-    """Merge multiple BIOM tables into one.
+    Parameters:
+    -----------
+    input_dir : string
+        name of the directory containing the deblurred fasta files
+    file_end : string
+        the ending of all the fasta files to be added to the table
+        (default '.fasta.trim.derep.no_artifacts.msa.deblur.no_chimeras')
 
-    Code taken from QIIME's merge_otu_tables.py
+    Returns
+    -------
+    names : list of tuples of (string,string)
+        list of tuples of:
+            name of fasta files to be added to the biom table
+            sampleid (file names without the file_end and path)
+    """
+    logger = logging.getLogger(__name__)
+    logger.debug('get_files_for_table input dir %s, '
+                 'file-ending %s' % (input_dir, file_end))
+
+    names = []
+    # scan the input directory entries
+    for cfile in listdir(input_dir):
```

Comment:
@amnona Sorry for not mentioning this earlier, I just remember about glob. I think this entire `for` loop can be simplified by using this code:

``` python
from os.path import join
from glob import glob
for cfile in glob(join(input_dir, "*%s" % file_end)):
    if not isfile(cfile):
        continue
    sample_id = cfile[:-len(file_end)]
    names.append((cfile, sampleid))
```

`glob` should already return all the paths that match the `"*.fasta.trim.derep.no_artifacts.msa.deblur.no_chimeras"` wildcard on the given directory. I tested it on a directory that I have and seems to do the same here, worth doublechecking though.
Also `cfile` will be the correct path already, so you will not need the join.